src/nupic/support/__init__.py

```diff
@@ -431,13 +431,18 @@ def makeKey(name):
     replacements[makeKey('SYSLOG_HANDLER_ADDRESS')] = '"/var/run/syslog"'
   elif platform.startswith('linux'):
     replacements[makeKey('SYSLOG_HANDLER_ADDRESS')] = '"/dev/log"'
+  elif platform.startswith('win'):
+    replacements[makeKey('SYSLOG_HANDLER_ADDRESS')] = '"log"'
```

Comment:
@scottpurdy: there are a few things going on here that are not quite right. 

1. It's better to remove the SYSLOG_HANDLER altogether from this script and from nupic-logging.conf, because we actually don't make use of it due to critical limitations. At some point, we experimented with SysLogHandler on Linux and Mac OS X. We found that it didn't work  with long log lines that we often need for debugging, such as capturing a deep traceback. We experimented and found that the two platforms differed in how they handled "overflow": one of them would drop the log message, while the other would truncate at the limit. So, we decided against using them. We must have forgotten to remove the SYSLOG_HANDLER references from this script and from nupic-logging.conf. You can see that `nupic.support.initLogging()` doesn't actually add SysLogHandler to the `handlers` list, so it's not going to be used.
2. SysLogHandler is not supported on windows, but the above line recommends to get rid of this stuff, anyway. If we really need to support SysLogHandler and NTEventLogHandler-like functionality, then we would need an additional effort to make it work right. If there is no such requirement, let's just drop this SysLogHandler/NEventLogHandler functionality altogether.
 ```
  if platform.startswith('darwin'):
    replacements[makeKey('SYSLOG_HANDLER_ADDRESS')] = '"/var/run/syslog"'
  elif platform.startswith('linux'):
    replacements[makeKey('SYSLOG_HANDLER_ADDRESS')] = '"/dev/log"'
  elif platform.startswith('win'):
    replacements[makeKey('SYSLOG_HANDLER_ADDRESS')] = '"log"'
  else:
    raise RuntimeError("This platform is neither darwin, win32, nor linux: %s" % (
      sys.platform,))
```
3. The first block of code below was intended to initialize file logging in the suppressed state; hence, `/dev/null`. However, the `'win'` initialization doesn't match that intent and needs to be fixed. It's only in the `if configLogDir is not None:` block that we should be enabling file logging.
 ```
  # Nupic logs go to file
  replacements[makeKey('PERSISTENT_LOG_HANDLER')] = 'fileHandler'
  if platform.startswith('win'):
    replacements[makeKey('FILE_HANDLER_LOG_FILENAME')] = '"nupic.log"'
  else:
    replacements[makeKey('FILE_HANDLER_LOG_FILENAME')] = '"/dev/null"'

  # Set up log file path for the default file handler and configure handlers
  handlers = list()
  
  if configLogDir is not None:
    logFilePath = _genLoggingFilePath()
    makeDirectoryFromAbsolutePath(os.path.dirname(logFilePath))
    replacements[makeKey('FILE_HANDLER_LOG_FILENAME')] = repr(logFilePath)

    handlers.append(replacements[makeKey('PERSISTENT_LOG_HANDLER')])
```